### PR TITLE
Fix Windows Build: Disable nodiscard warning.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libevmasm/KnownState.h [hunk 1]
```diff
@@ -37,8 +37,17 @@
 #pragma clang diagnostic ignored "-Wredeclared-class-member"
 #endif // defined(__clang__)
 
+#if defined(_MSC_VER)
```
Comment (bshastry) [inline libevmasm/KnownState.h:41]: Perhaps add a comment here?

```
This silences the nodiscard warning in `boost/bimap.hpp` that appeared post automated migration from Circle CI MSVC version 19.28.29337.0 to version 19.28.29913.0.
```

**Review by bshastry — COMMENTED** [review]
Perhaps add a comment to document the issue.

File: libevmasm/KnownState.h (hunk 1 above)
Comment (chriseth) [reply libevmasm/KnownState.h:41]: In addition to the "nodiscard", we might add "try removing it again at some point".

**Comment by chriseth** [comment]
@bshastry better like this?

**Review by bshastry — APPROVED** [review]
LGTM if CIs pass.

**Pull request merged**